src/core/json/driver.cc

```diff
@@ -58,6 +61,25 @@ class MaxImpl : public SingleValueImpl {
 
     return true;
   }
+
+  bool ApplyImpl(flexbuffers::Reference src) final {
+    if (!src.IsNumeric()) {
+      return false;
+    }
+
+    visit(Overloaded{
+              [&](monostate) { Init(src); },
+              [&](double d) { val_ = max(d, src.AsDouble()); },
+              [&](int64_t i) {
+                if (src.IsFloat())
+                  val_ = max(double(i), src.AsDouble());
+                else
+                  val_ = max(i, src.AsInt64());
```

Comment:
nit: val = src.IsFloat() ? max(double(i), src.AsDouble()) : max(i, src.AsInt64());

Reply:
ternary operators must return the same type from both branches.

Reply:
val = src.IsFloat() ? max(double(i), src.AsDouble()) : Result(max(i, src.AsInt64()));

Reply:
you can skip it if you want